### PR TITLE
Helm ga

### PR DESCRIPTION
**Pull request by dan-mckean**
A number of commits making changes to different pages in relation to the Helm GA. 

## Test plan
Docs

### CODE REVIEW

File: doc/admin/external_services/redis.md [hunk 1]
```diff
@@ -18,6 +19,11 @@ or follow the [IANA specification for Redis URLs](https://www.iana.org/assignmen
 
 If using Docker for Desktop, `host.docker.internal` will resolve to the host IP address.
 
-## Kubernetes Redis
- - See our documentation for Kubernetes [here](../install/kubernetes/configure.md#configure-custom-redis)
+## Kubernetes
+
+### Kubernetes with Helm
+[See the Helm Redis guidance here](../install/kubernetes/helm#using-external-redis-instances)
```
Comment (dan-mckean) [inline doc/admin/external_services/redis.md:25]:
```suggestion
[See the Helm Redis guidance here](../install/kubernetes/helm.md#using-external-redis-instances)
```

File: doc/admin/updates/kubernetes.md [hunk 2]
```diff
@@ -16,7 +16,9 @@ The`codeinsights-db` container no longer uses TimescaleDB and is now based on th
 
 **CAUTION** If you use a custom Code Insights postgres config, you must update the `shared_preload_libraries` list to remove timescaledb. The [above PR](https://github.com/sourcegraph/deploy-sourcegraph/pull/4103/files#diff-e5f8d6e46f8c9335c489c0d8e9ae9be4f4655f878f3ac569c73ebb3865b0eeeeL695-R688) demonstrates this change.
 
-Follow the [standard upgrade procedure](../install/kubernetes/update.md) to upgrade your deployment.
+To upgrade your deployment follow either:
+  * [Kubernetes with Helm upgrade guide](../install/kubernetes/helm#upgrading-sourcegraph)
```
Comment (dan-mckean) [inline doc/admin/updates/kubernetes.md:20]:
```suggestion
  * [Kubernetes with Helm upgrade guide](../install/kubernetes/helm.md#upgrading-sourcegraph)
```

File: doc/adopt/enterprise_getting_started_guide.md [hunk 3]
```diff
@@ -26,15 +26,11 @@ If you're deploying a new Enterprise instance, this page covers our most frequen
...
-- [Provisioning a Kubernetes cluster](../admin/install/kubernetes/configure.md#configuring-sourcegraph)
-- [Amazon EKS](../admin/install/kubernetes/eks.md)
-- [Amazon EC2](https://kubernetes.io/docs/setup/)
-- [Google Kubernetes Engine](https://cloud.google.com/kubernetes-engine/docs/quickstart)
-- [Azure](../admin/install/kubernetes/azure.md)
-- [Scaling](../admin/install/kubernetes/scale.md#improving-performance-with-a-large-number-of-repositories) 
-- [Ingress controller](../admin/install/kubernetes/configure.md#ingress-controller-recommended)
-- [Common configuration](../admin/install/kubernetes/configure.md#common-configuration)
+- [All Kubernetes with Helm guidance](../admin/install/kubernetes/helm.md)
+- [Amazon EKS](../admin/install/kubernetes/helm#configure-sourcegraph-on-elastic-kubernetes-service-eks)
```
Comment (dan-mckean) [inline doc/adopt/enterprise_getting_started_guide.md:30]:
```suggestion
- [Amazon EKS](../admin/install/kubernetes/helm.md#configure-sourcegraph-on-elastic-kubernetes-service-eks)
```

File: doc/adopt/enterprise_getting_started_guide.md [hunk 4]
```diff
@@ -26,15 +26,11 @@ If you're deploying a new Enterprise instance, this page covers our most frequen
...
+- [Google GKE](../admin/install/kubernetes/helm#configure-sourcegraph-on-google-kubernetes-engine-gke)
```
Comment (dan-mckean) [inline doc/adopt/enterprise_getting_started_guide.md:31]:
```suggestion
- [Google GKE](../admin/install/kubernetes/helm.md#configure-sourcegraph-on-google-kubernetes-engine-gke)
```

File: doc/adopt/enterprise_getting_started_guide.md [hunk 5]
```diff
@@ -26,15 +26,11 @@ If you're deploying a new Enterprise instance, this page covers our most frequen
...
+- [Azure](../admin/install/kubernetes/helm#configure-sourcegraph-on-azure-managed-kubernetes-service-aks)
```
Comment (dan-mckean) [inline doc/adopt/enterprise_getting_started_guide.md:32]:
```suggestion
- [Azure](../admin/install/kubernetes/helm.md#configure-sourcegraph-on-azure-managed-kubernetes-service-aks)
```

File: doc/adopt/enterprise_getting_started_guide.md [hunk 6]
```diff
@@ -26,15 +26,11 @@ If you're deploying a new Enterprise instance, this page covers our most frequen
...
+- [Configure Sourcegraph on other Cloud providers or on-prem](../admin/install/kubernetes/helm#configure-sourcegraph-on-other-cloud-providers-or-on-prem)
```
Comment (dan-mckean) [inline doc/adopt/enterprise_getting_started_guide.md:33]:
```suggestion
- [Configure Sourcegraph on other Cloud providers or on-prem](../admin/install/kubernetes/helm.md#configure-sourcegraph-on-other-cloud-providers-or-on-prem)
```

File: doc/admin/external_services/redis.md [hunk 7]
```diff
@@ -18,6 +19,11 @@ or follow the [IANA specification for Redis URLs](https://www.iana.org/assignmen
...
+
+### Kebernetes without Helm
```
Comment (kevinwojo) [inline doc/admin/external_services/redis.md:27]:
```suggestion
### Kubernetes without Helm
```

File: doc/admin/install/index.md [hunk 8]
```diff
@@ -16,19 +16,23 @@ If you're just starting out, you can [**try Sourcegraph Cloud**](https://sourceg
...
+> NOTE: Kubernetes with Helm deployment times vary based on the level and complexity of customizations you require.
+
+> WARNING: Some features for self-hosted deployments [require a Sourcegraph license](https://about.sourcegraph.com/pricing/).
 
 ### Tips
 
-* **We recommend Docker Compose for most initial production deployments**. You can [migrate to a different deployment method](../updates/index.md#migrating-to-a-new-deployment-type) later on if needed.
-* Note that **for a Kubernetes deployment, you are expected to have a team that is familiar with operating Kubernetes clusters**, including but not limited to the use of persistent storage. If in any doubt about your team's ability to support this, please speak to your Sourcegraph contact about using Docker Compose instead.
-* Don't want to worry about managing a Sourcegraph deployment? Consider a [managed instance](./managed.md).
+* **We recommend Kubernetes with Helm for most  production deployments**.
```
Comment (kevinwojo) [inline doc/admin/install/index.md:31]:
```suggestion
* **We recommend Kubernetes with Helm for most production deployments**.
```

**Review by kevinwojo — APPROVED** [review]

File: doc/admin/install/index.md [hunk 9]
```diff
@@ -16,19 +16,23 @@ If you're just starting out, you can [**try Sourcegraph Cloud**](https://sourceg
...
 | ----------------------------------------------------------- | ------------------------------------------------------- | ----------------- | :----------------: | :----------: | :-----------: |
-| [**★ Docker Compose**](../install/docker-compose/index.md) | **Small & medium** production deployments               | 🟢 5 minutes     |         ✅         |      ✅      |      ❌       |
-| [**★ Kubernetes**](../install/kubernetes/index.md)         | **Medium & large** highly-available cluster deployments | 🟠 30-90 minutes |         ✅         |      ✅      |      ✅       |
-| [Single-container](../install/docker/index.md)              | Local testing                                           | 🟢 1 minute      |         ❌         |      ❌      |      ❌       |
+| [**★ Kubernetes with Helm **](../install/kubernetes/helm.md)         | Production deployments of any size | 5-90 minutes |         ✅         |      ✅      |      ✅       |
+| [** Docker Compose**](../install/docker-compose/index.md) | Production deployments where Kubernetes with Helm is not viable               | 5 - 30 minutes     |         ✅         |      ✅      |      ❌       |
```
[the remark this answers is not in the document]
Comment (dan-mckean) [reply doc/admin/install/index.md:20]: I don't think existing Docker compose users will notice. And if they do, it's not like we're saying it's unsupported. And the factors in the table haven't really changed (except I cut the minimum time for Kubernetes) so it's not like they're losing out before.
We're also not saying "Migrate asap"!

As for docker-compose being suitable... is using Helm a worse choice if they can go that route?

File: doc/admin/install/index.md [hunk 10]
```diff
@@ -16,19 +16,23 @@ If you're just starting out, you can [**try Sourcegraph Cloud**](https://sourceg
 
 | Deployment Type                                             | Suggested for                                           | Setup time        | Resource isolation | Auto-healing | Multi-machine |
 | ----------------------------------------------------------- | ------------------------------------------------------- | ----------------- | :----------------: | :----------: | :-----------: |
-| [**★ Docker Compose**](../install/docker-compose/index.md) | **Small & medium** production deployments               | 🟢 5 minutes     |         ✅         |      ✅      |      ❌       |
-| [**★ Kubernetes**](../install/kubernetes/index.md)         | **Medium & large** highly-available cluster deployments | 🟠 30-90 minutes |         ✅         |      ✅      |      ✅       |
-| [Single-container](../install/docker/index.md)              | Local testing                                           | 🟢 1 minute      |         ❌         |      ❌      |      ❌       |
+| [**★ Kubernetes with Helm **](../install/kubernetes/helm.md)         | Production deployments of any size | 5-90 minutes |         ✅         |      ✅      |      ✅       |
```
[the remark this answers is not in the document]
Comment (dan-mckean) [reply doc/admin/install/index.md]: I've added a line back in for it @caugustus-sourcegraph, and added a new column. Take a look and see what you think?

**Review by caugustus-sourcegraph — APPROVED** [review]

**Pull request merged**